### DISCUSSION TITLE
How request usage is calculated

### DISCUSSION THREAD

**Discussion by giovannicandido** [zitadel/zitadel#6347] [General]
Hi, I start an evaluation of the cloud offering. I used it a bit to create a test user and project, did some authentications in the console and in a test app, and did some management operations. 
In 12 hours my request count is 707 which I think is high for such a light interaction.
That means on the same usage in 20 days I could have 14140 requests, and I do not even have users in production, and not even tested the service appropriately.
So I would like to know how requests are counted because the site states an estimated calculation on prices that show much higher users per 25000 requests.

**Comment by mffap** [comment]
You can find the definition of authenticated requests here: https://zitadel.com/docs/legal/cloud-service-description

Example based on that definition:
When a user logs in with the hosted login, then no requests are being counted until a valid token is issued (1 request). You may pull the user info (1 request) as a subsequent action. When you refresh the token thats an additional request.

I assume you used the Console to configure your setup. Console is built on top of our APIs and is quite hungry, as it pulls in much context information (eg, last events of an object, org switcher, managers, etc.). This results in a high usage when clicking around. Your end-users will typically not interact with Console, but you probably integrate the most important parts into your application.

Hope this helps. Maybe try out a real flow with one of our [example apps](https://github.com/search?q=topic%3Aexamples+org%3Azitadel&type=repositories)  and see how much usage you get.
